engine/src/main/scala/cromwell/services/MetadataServiceActor.scala

```diff
@@ -63,15 +63,40 @@ object MetadataServiceActor {
 }
 
 case class MetadataServiceActor(serviceConfig: Config, globalConfig: Config) extends Actor {
+
+  private var journal = List.empty[MetadataEvent]
+
   def receive = {
-    case PutMetadataAction(event) =>
-      val result: MetadataPutAcknowledgement = ???
-      sender ! result
+    case action@PutMetadataAction(event) =>
+      journal = journal :+ event
+      sender ! MetadataPutAcknowledgement(action)
     case GetAllMetadataAction(workflowId) =>
-      val result: MetadataLookupResponse = ???
-      sender ! result
-    case GetMetadataQueryAction(MetadataQuery(workflowId, jobKey, key)) =>
-      val result: MetadataLookupResponse = ???
-      sender ! result
+      val query = MetadataQuery(Option(workflowId), None, None)
+      val filtered = filterByQuery(query)
+      sender ! MetadataLookupResponse(query, filtered)
+    case GetMetadataQueryAction(query@MetadataQuery(workflowId, jobKey, key)) =>
```

Comment:
Not a strong opinion here, but `workflowId`, `jobKey` and `key` are all unused here and probably good candidates for a wildcard `_` to emphasize it's non-usefulness.